detox/test/e2e/02.matchers.test.js

```diff
@@ -30,8 +32,8 @@ describe('Matchers', () => {
 
   // https://facebook.github.io/react-native/docs/accessibility.html#accessibilitytraits-ios
   // Accessibility Inspector in the simulator can help investigate traits
-  it(':ios: should match elements by accesibility trait', async () => {
-    await element(by.traits(['button', 'text'])).tap();
+  it(':ios: should match elements by accessibility trait', async () => {
+    await element(by.traits(['button', 'disabled'])).tap();
```

Comment:
I don't understand this change. If the traits are a button and disabled, how can you tap it? That's not how accessibility traits work.

Reply:
This is a straight swap for what the test was doing originally. The accessibility trait of 'text' would have actually corresponded to 'disabled'

Reply:
I don't think this is true, because a button also has text. If an accessibility element is disabled, it should not be interactive.

Reply:
OK. Well something is misaligned somewhere. If I keep the value as `text` the test correctly finds elements with a `text` trait but fails as there are more than one.

Reply:
https://developer.apple.com/documentation/uikit/uiaccessibilitytraitnotenabled?language=objc

> The accessibility element is not enabled and does not respond to user interaction.

Reply:
I don't dispute that. But as you can see the `text` trait would correspond to the value of 256 which should be `disabled` so does that fact that this test passes suggest a problem elsewhere.

Reply:
If you search for button+text (with the correct values), which element is selected? Likewise, if you search for button+disabled (with the correct values), which element is selected?

Reply:
With button+text (corrected values) I get ```  1) Matchers
       :ios: should match elements by accessibility trait:
     Error: Error: Multiple UI elements matched for the given criteria.
Exception with Action: {
  "Action Name":  "Tap",
  "Element Matcher":  "((!(kindOfClass('RCTScrollView')) && ((respondsToSelector(isAccessibilityElement) && isAccessibilityElement) && accessibilityTraits: UIAccessibilityTraitButton,UIAccessibilityTraitStaticText)) || (((kindOfClass('UIView') || respondsToSelector(accessibilityContainer)) && parentThatMatches(kindOfClass('RCTScrollView'))) && ((kindOfClass('UIView') || respondsToSelector(accessibilityContainer)) && parentThatMatches(((respondsToSelector(isAccessibilityElement) && isAccessibilityElement) && accessibilityTraits: UIAccessibilityTraitButton,UIAccessibilityTraitStaticText)))))",
  "Recovery Suggestion":  "Create a more specific matcher to uniquely match an element. If that's not possible then use atIndex: to select from one of the matched elements but the order of elements may change."
}
```

Reply:
```
Error Trace: [
  {
    "Description":  "Multiple elements were matched: (
    "<UILabel:0x7fe33b520410; AX=Y; AX.label='Active'; AX.frame={{147.16666666666666, 26}, {80.666666666666657, 36}}; AX.activationPoint={187.5, 44}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{147.16666666666666, 26}, {80.666666666666671, 36}}; opaque; alpha=1; UIE=N; text='Active'>",
    "<RCTView:0x7fe33d9350b0; AX=Y; AX.label='Traits'; AX.frame={{169.33332824707031, 288.33334350585938}, {36.333343505859375, 27}}; AX.activationPoint={187.5, 301.83334350585938}; AX.traits='UIAccessibilityTraitButton,UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{169.33332824707031, 288.33334350585938}, {36.333343505859375, 27}}; opaque; alpha=1>",
    "<RCTTextView:0x7fe33d938d90; AX=Y; AX.label='Index'; AX.frame={{170, 580.66668701171875}, {35.666656494140625, 17.3333740234375}}; AX.activationPoint={187.83332824707031, 589.3333740234375}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{0, 0}, {35.666656494140625, 17.3333740234375}}; alpha=1>",
    "<RCTTextView:0x7fe33d938680; AX=Y; AX.label='Index'; AX.frame={{170, 543.66668701171875}, {35.666656494140625, 17.3333740234375}}; AX.activationPoint={187.83332824707031, 552.3333740234375}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{0, 0}, {35.666656494140625, 17.3333740234375}}; alpha=1>",
    "<RCTTextView:0x7fe33d937f70; AX=Y; AX.label='Index'; AX.frame={{170, 506.66665649414062}, {35.666656494140625, 17.333343505859375}}; AX.activationPoint={187.83332824707031, 515.33332824707031}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{0, 0}, {35.666656494140625, 17.333343505859375}}; alpha=1>",
    "<RCTTextView:0x7fe33d937860; AX=Y; AX.id='ProductId003'; AX.label='Product'; AX.frame={{268.66666793823242, 459.33334350585938}, {51.66668701171875, 17.33331298828125}}; AX.activationPoint={294.5000114440918, 468}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{224, 10}, {51.66668701171875, 17.33331298828125}}; alpha=1>",
    "<RCTTextView:0x7fe33d937450; AX=Y; AX.id='ProductId002'; AX.label='Product'; AX.frame={{197.33333969116211, 459.33334350585938}, {51.666671752929688, 17.33331298828125}}; AX.activationPoint={223.16667556762695, 468}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{152.66667175292969, 10}, {51.666671752929688, 17.33331298828125}}; alpha=1>",
    "<RCTTextView:0x7fe33d937040; AX=Y; AX.id='ProductId001'; AX.label='Product'; AX.frame={{126.00000381469727, 459.33334350585938}, {51.666671752929688, 17.33331298828125}}; AX.activationPoint={151.83333969116211, 468}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{81.333335876464844, 10}, {51.666671752929688, 17.33331298828125}}; alpha=1>",
    "<RCTTextView:0x7fe33d936c30; AX=Y; AX.id='ProductId000'; AX.label='Product'; AX.frame={{54.666667938232422, 459.33334350585938}, {51.666667938232422, 17.33331298828125}}; AX.activationPoint={80.500001907348633, 468}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{10, 10}, {51.666667938232422, 17.33331298828125}}; alpha=1>",
    "<RCTTextView:0x7fe33d934960; AX=Y; AX.label='Traits'; AX.frame={{169.33332824707031, 288.33334350585938}, {36.333343505859375, 17.33331298828125}}; AX.activationPoint={187.5, 297}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{0, 0}, {36.333343505859375, 17.33331298828125}}; alpha=1>",
    "<RCTTextView:0x7fe33d924140; AX=Y; AX.id='UniqueId345'; AX.label='ID'; AX.frame={{180.66667175292969, 251.33332824707031}, {14.333343505859375, 17.333328247070312}}; AX.activationPoint={187.83334350585938, 259.99999237060547}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{0, 0}, {14.333343505859375, 17.333328247070312}}; alpha=1>",
    "<RCTTextView:0x7fe33d91e1a0; AX=Y; AX.label='Label'; AX.frame={{169.66667175292969, 214.33332824707031}, {35.666656494140625, 17.333343505859375}}; AX.activationPoint={187.5, 223}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{0, 0}, {35.666656494140625, 17.333343505859375}}; alpha=1>"
). Please use selection matchers to narrow the selection down to single element.",
    "Error Domain":  "com.google.earlgrey.ElementInteractionErrorDomain",
    "Error Code":  "5",
    "File Name":  "GREYElementInteraction.m",
    "Function Name":  "-[GREYElementInteraction grey_errorForMultipleMatchingElements:withMatchedElementsIndexOutOfBounds:]",
    "Line":  "956"
  }
```

Reply:
I don't understand what is going on here. Only one of these is a button in addition to text: `RCTView:0x7fe33d9350b0`. I don't understand why it finds the rest.

Reply:
And none of them are disabled.

Reply:
OK, I see the problem. User Accessibility Inspector to check out all the buttons there, and indeed they are all text+button. So it finds them all.

Could you please disable the test? It's a bad test. Thanks

Reply:
Do you mean disable (with `it.skip`) or delete?

Reply:
Skip. Hopefully, we'll look at it eventually and make it a test that makes sense.